src/main/java/org/eclipsefoundation/marketplace/model/QueryParams.java

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Wrapper class for query parameter functionality, wrapping a Map of String to
+ * collection of String values. This class should be used anywhere where query
+ * parameters are used to control access and prevent bad data from being
+ * entered.
+ * 
+ * @author Martin Lowe
+ *
+ */
+public class QueryParams {
+	private static final String EMPTY_KEY_MESSAGE = "Key must not be null or blank";
+
+	private Map<String, List<String>> params;
+
+	/**
+	 * Builds the internal map using a copy of the data of the passed parameter map.
+	 * The map must exist to be properly set.
+	 * 
+	 * @param params list of parameters to use in the query map.
+	 */
+	public QueryParams(Map<String, List<String>> params) {
+		Objects.requireNonNull(params);
+
+		this.params = new HashMap<>(params);
+	}
+
+	/**
+	 * Retrieves the first value set in a list from the map for a given key.
+	 * 
+	 * @param params the parameter map containing the value
+	 * @param key    the key to retrieve the value for
+	 * @return the first value set in the parameter map for the given key, or null
+	 *         if absent.
+	 */
+	public String getFirstParam(String key) {
```

Comment:
Instead of making null a potential return value, I'd suggest to use the Optional<String> type as the return value. Return Optional.empty() instead of null and Optional.of(vals.get(0)) otherwise.